go-controller/pkg/kubevirt/pod.go

```diff
@@ -503,16 +507,71 @@ func ReconcileIPv4DefaultGatewayAfterLiveMigration(watchFactory *factory.WatchFa
 	}
 
 	lrpMAC := util.IPAddrToHWAddr(lrpJoinIPv4)
-	for _, subnet := range netInfo.Subnets() {
+	for _, subnet := range r.netInfo.Subnets() {
 		gwIP := util.GetNodeGatewayIfAddr(subnet.CIDR).IP.To4()
 		if gwIP == nil {
 			continue
 		}
 		garp := util.GARP{IP: gwIP, MAC: &lrpMAC}
-		if err := util.BroadcastGARP(interfaceName, garp); err != nil {
+		if err := util.BroadcastGARP(r.interfaceName, garp); err != nil {
 			return err
 		}
 	}
-
 	return nil
 }
+
+// ReconcileIPv6AfterLiveMigration will do two things at VM's:
+// - Remove ipv6 default gw path from VM's node before live migration
+// - Add ipv6 default gw path from VM's node after live migration
+// This is done by sending a pair of unsolicited RA's one with lifetime=0
+// (to remove the gateway path) another with lifetime=max to add the new
+// default gateway path
+func (r *DefaultGatewayReconciler) ReconcileIPv6AfterLiveMigration(liveMigration *LiveMigrationStatus) error {
+	if liveMigration.State != LiveMigrationTargetDomainReady {
+		return nil
+	}
+	nodes, err := r.watchFactory.GetNodes()
+	if err != nil {
+		return err
+	}
+	ras := make([]ndp.RouterAdvertisement, 0, len(nodes))
+	for _, node := range nodes {
```

Comment:
while I don't see anything wrong w/ this approach, I think the code would be more readable if it looked something like (using some sort of pseudo-code):
```golang
targetNode := migrationTargetNode(liveMigration)
nonTargetNodes := nonTargetNodes(liveMigration)

ras := []ndp.RouterAdvertisement{configNewGateway(targetNode), forgetOldGateways(nonTargetNodes)...}
...
```

Or something along those lines. This is highly opinionated, feel free to disregard.

Reply:
Don't know if it's worth it for temporal code like this.